File: solutions/_2264_largest_3_same_digit_number_in_string/solution.cpp
#include <string>
#include <vector>
using namespace std;

class Solution
{
public:
    string largestGoodInteger(string num)
    {
        string largest = "";
        char ch;
        for (int i = 0; i <= num.size() - 3; i++)
        {
            ch = num[i];
            if (ch == num.at(i + 1) && ch == num.at(i + 2))
            {
                if (largest.empty() || num.substr(i, 3) > largest)
                {
                    largest = num.substr(i, 3);
                }
            }
        }
        return largest;
    }
};

/*
https://leetcode.com/problems/largest-3-same-digit-number-in-string/
*/

File: solutions/_2264_largest_3_same_digit_number_in_string/solution.py
class Solution:
    def largestGoodInteger(self, num: str) -> str:
        largest = ""
        for i in range(len(num) - 2):
            if num[i] == num[i + 1] == num[i + 2]:
                if num[i] > largest:
                    largest = num[i] * 3
        return largest


# https://leetcode.com/problems/largest-3-same-digit-number-in-string/
